backend/courses/util.py

```diff
@@ -506,6 +512,73 @@ def upsert_course_from_opendata(info, semester, missing_sections=None):
         missing_sections.discard(section.full_code)
 
 
+def add_attributes(section, attributes):
+    """
+    Add attributes to course of section.
+    Create attribute if it does not exist
+    """
+    for attribute in attributes:
+        try:
+            attr = Attribute.objects.get(
+                code=attribute.get("attribute_code"),
+            )
+        except ObjectDoesNotExist:
+            school = identify_school(attribute.get("attribute_code"))
+            attr = Attribute.objects.create(
+                code=attribute.get("attribute_code"),
+                description=attribute.get("attribute_desc"),
+                school=school,
+            )
+        attr.courses.add(section.course)
+
+
+def identify_school(attribute_code):
```

Comment:
If we get more structured degree requirements data from Charles Harvey (see [this thread](https://penn-labs.slack.com/archives/CDDK7CB53/p1653062544925119)), maybe we can improve these heuristics. For now this looks good though.

Reply:
Sounds good!